src/filelock/_soft.py

```diff
@@ -12,14 +12,17 @@ class SoftFileLock(BaseFileLock):
     def _acquire(self) -> None:
         raise_on_exist_ro_file(self._lock_file)
         # first check for exists and read-only mode as the open will mask this case as EEXIST
-        mode = (
+        flags = (
             os.O_WRONLY  # open for writing only
             | os.O_CREAT
             | os.O_EXCL  # together with above raise EEXIST if the file specified by filename exists
             | os.O_TRUNC  # truncate the file to zero byte
         )
+        mode = 0o660
+        umask = 0o002
```

Comment:
After looking into this seems chmod/umask is not supported on Windows. Calling those methods will not fail but Windows does not honour those settings as the underlying filesystem does not understand it. In light of that can we not set these on Windows? My proposal is to move the mode/umask shenanigans into a context manager here and not call it on Windows.

Reply:
I also did some research and tried several things on Windows via VirtualBox. File permissions are completely different and have nothing to do with those under Linux. I even asked on StackOverflow and got some hints:

https://stackoverflow.com/questions/69782278/os-open-windows-file-permissions-mode?noredirect=1#comment123348708_69782278

> My proposal is to move the mode/umask shenanigans into a context manager here and not call it on Windows.

I'm not sure that I understand your point. There already is a `_windows.py` with this:

```
if sys.platform == "win32":  # pragma: win32 cover
    import msvcrt

    class WindowsFileLock(BaseFileLock):
```

And this happens in `__init__.py`:

```
if sys.platform == "win32":  # pragma: win32 cover
    _FileLock: Type[BaseFileLock] = WindowsFileLock
```

Why would the setting of `umask+mode` have to be moved away? I'm still new to this project, so is there something I've overlooked?

Reply:
I was talking about the soft filelock implementation solely, not the platform specific locks.

Reply:
I'll look into it, although it might take some time.